### PR TITLE
Use go test directly with flags instead of make e2e

### PR DESCRIPTION
**Pull request by yutongz**

### CODE REVIEW

File: prow/daily-release-qualification.sh [hunk 1]
```diff
@@ -63,5 +63,18 @@ cp -R ${DAILY_BUILD}/install/* install/
...
 
 echo 'Running E2E Tests'
-ARGS="--test_logs_path="${ARTIFACTS_DIR}" --istioctl_url "${ISTIOCTL_STAGE_URL}" "$@""
-make e2e_all E2E_ARGS="${ARGS}"
+ARGS=( --test_logs_path="${ARTIFACTS_DIR}" --istioctl_url "${ISTIOCTL_STAGE_URL}" "$@" )
+
+EXTRA_E2E_ARGS=()
+EXTRA_E2E_ARGS+=( --mixer_tag="${TAG}" )
+EXTRA_E2E_ARGS+=( --pilot_tag="${TAG}" )
+EXTRA_E2E_ARGS+=( --proxy_tag="${TAG}")
```
Comment (chxchx) [inline prow/daily-release-qualification.sh]: nit: add a white space before closing parenthesis to be consistent in style

File: prow/daily-release-qualification.sh [hunk 2]
```diff
@@ -63,5 +63,18 @@ cp -R ${DAILY_BUILD}/install/* install/
...
+EXTRA_E2E_ARGS+=( --ca_tag="${TAG}")
+EXTRA_E2E_ARGS+=( --mixer_hub="${HUB}")
+EXTRA_E2E_ARGS+=( --pilot_hub="${HUB}")
+EXTRA_E2E_ARGS+=( --proxy_hub="${HUB}")
+EXTRA_E2E_ARGS+=( --ca_hub="${HUB}")
+
+go test -v -timeout 20m ./tests/e2e/tests/simple -args ${ARGS[@]} ${EXTRA_E2E_ARGS[@]}
+go test -v -timeout 20m ./tests/e2e/tests/mixer -args ${ARGS[@]} ${EXTRA_E2E_ARGS}
```
Comment (chxchx) [inline prow/daily-release-qualification.sh]: how is `${EXTRA_E2E_ARGS[@]}` different from `${EXTRA_E2E_ARGS}` (line 78 79)? Can we choose one of them for all go test commands?

**Comment by mattdelco** [comment]
I know nearly nothing about these tests, but if this is about the daily release failures then I'm left wondering if this approach is going after the symptom rather than the problem.  I thought the problem might be related to using locally compiled artifacts rather than already-built ones, and I'm not sure if all these flags are doing runtime override of what's built-in to the wrong artifacts.  Stated different, the "default" tests don't seem to have much room to use a default when everything is specified.    I thought about doing an experiment where all the e2e_XXX targets no longer depend on "istioctl generate_yaml", thus making it more obvious where there might be inappropriate use of locally-compiled artifacts.

As for replacing a make target with go commands, this might be going in the opposite direction of what I've seen people discuss in CircleCI changes.  I've seen sentiment expressed that there should be make targets that provide most of the needed parameters to "go test".  The idea is to have fewer parameters specified in .circleci/config.yml to be passed to make.  I think part of the motivation is to get things more in common between CircleCI and Prow (e.g., if a parameter gets added or changed then it only needs to be updated in the Makefile rather than separately in both CircleCI and Prow).

File: prow/daily-release-qualification.sh (hunk 1 above)
Comment (yutongz) [reply prow/daily-release-qualification.sh]: Done

File: prow/daily-release-qualification.sh (hunk 2 above)
Comment (yutongz) [reply prow/daily-release-qualification.sh]: Nice catch, they all should be with [@].

Only `${EXTRA_E2E_ARGS[@]}` can give you all elements in the array.

**Comment by yutongz** [comment]
@mattdelco 

I am trying to feed the right artifacts to the e2e framework, sorry not quiet familiar about the build system. So

1. Today's problem is because we specify both `--istioctl` (which is local binary built by `make istioctl`) and `--istioctl_url` (which told e2e framework to download one from a remote place). And I believe the in normal dev-process, building one on the fly and use it in test makes sense. But in our release qualification, we want to get one (and specifically the stage one I am pointing to in https://github.com/istio-releases/daily-release/pull/350) from gcs where build target is. And unfortunately, `istioctl` will override `istioctl_url` 

2. While since I am not quiet sure what will affect if I change makefile https://github.com/istio/istio/blob/master/tests/istio.mk#L50, I would prefer separate the dev-process with the release-process and make it clear what flag we are using (what version we are testing)

3. And the default tests, we use https://github.com/istio-releases/daily-release/blob/master/prow/daily-e2e-rbac-auth-default.sh#L29. And it's handled in e2e framework https://github.com/istio/istio/blob/master/tests/e2e/framework/istioctl.go#L142. With `--default_proxy`, it calls `istioctl kube-inject` without `--proxy_hub` or `--proxy_tag` specified.

**Comment by mattdelco** [comment]
/lgtm

I tend to think of $@ as being convenient in some cases but has some liability from making it harder to know what's passed through or needs to be supported (or is redundant/contradictory).  However, I don't plan to do anything about it (nor expect anyone else to either).

**Pull request merged**